src/go/rpk/pkg/cli/cmd/container/docker/client.go

```diff
@@ -0,0 +1,256 @@
+			Config: []network.IPAMConfig{
+				{
+					Subnet:  options.Subnet,
+					Gateway: options.Gateway,
+				},
+			},
+		},
+	}
+	return c.conn.NetworkCreate(ctx, name, nc)
+}
+
+func (c *DockerClient) NetworkRemove(ctx context.Context, networkID string) error {
+	return c.conn.NetworkRemove(ctx, networkID)
+}
+
+func (c *DockerClient) ContainerRemove(
+	ctx context.Context,
+	containerID string,
+	options types.ContainerRemoveOptions,
+) error {
+	return c.conn.ContainerRemove(ctx, containerID, options)
+}
+
+func (c *DockerClient) ContainerCreate(
+	ctx context.Context,
+	config *common.ContainerConfig,
+	hostConfig *common.ContainerHostConfig,
+	networkingConfig *common.ContainerNetwork,
+	containerName string,
+) (string, error) {
+	cc := &container.Config{
+		Image:        config.Image,
+		Hostname:     config.Hostname,
+		Cmd:          config.Cmd,
+		ExposedPorts: config.ExposedPorts,
+		Labels:       config.Labels,
+	}
+
+	hc := &container.HostConfig{
+		PortBindings: hostConfig.PortBindings,
+	}
+
+	nc := &network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{
+			networkingConfig.ID: {
+				IPAMConfig: &network.EndpointIPAMConfig{
+					IPv4Address: networkingConfig.IPAdress,
+				},
+				Aliases: networkingConfig.Aliases,
+			},
+		},
+	}
+	resp, err := c.conn.ContainerCreate(ctx, cc, hc, nc, nil, containerName)
+	if err != nil {
+		return "", err
+	}
+
+	return resp.ID, nil
+}
+
+func (c *DockerClient) ContainerStart(
+	ctx context.Context,
+	containerID string,
+) error {
+	return c.conn.ContainerStart(ctx, containerID, types.ContainerStartOptions{})
+}
+
+func (c *DockerClient) ContainerStop(
+	ctx context.Context,
+	containerID string,
+	timeout *time.Duration,
+) error {
+	return c.conn.ContainerStop(ctx, containerID, timeout)
+}
+
+func (c *DockerClient) ImageList(
+	ctx context.Context,
+	options common.ImageListOptions,
+) ([]types.ImageSummary, error) {
+	filters := filters.NewArgs(
+		filters.Arg("reference", options.Ref),
+	)
+
+	return c.conn.ImageList(ctx, types.ImageListOptions{
+		Filters: filters,
+	})
+}
+
+func (c *DockerClient) NetworkInspect(
+	ctx context.Context,
+	networkID string,
+) (common.NetworkInspect, error) {
+	net, err := c.conn.NetworkInspect(ctx, networkID, types.NetworkInspectOptions{})
+	if err != nil {
+		return common.NetworkInspect{}, err
+	}
+
+	if len(net.IPAM.Config) != 1 {
+		return common.NetworkInspect{}, fmt.Errorf(
+			"'%s' network config is corrupted",
+			net.Name,
+		)
+	}
+
+	ni := common.NetworkInspect{
+		Name:    net.Name,
+		Gateway: net.IPAM.Config[0].Gateway,
+	}
+
+	return ni, nil
+}
+
+func (c *DockerClient) NetworkList(
+	ctx context.Context,
+	options common.NetworkListOptions,
+) (common.NetworkList, error) {
+	args := filters.NewArgs()
+	args.Add("name", options.Name)
+
+	nl, err := c.conn.NetworkList(ctx, types.NetworkListOptions{Filters: args})
+	if err != nil {
+		return common.NetworkList{}, err
+	}
+
+	if len(nl) > 1 {
```

Comment:
Why this new check / what does this check help us ensure? Looks like the old code just looked for one string match. Is it possible to get more than one if we already have a filter on name (and if the network listing is not bugged)?